src/block/Slime.php

```diff
@@ -40,5 +42,19 @@ public function onEntityLand(Entity $entity) : ?float{
 		return -$entity->getMotion()->y;
 	}
 
-	//TODO: slime blocks should slow entities walking on them to about 0.4x original speed
+	public function onEntityWalking(Entity $entity) : void{
+        $motion = $entity->getMotion();
+        $entity->setMotion($motion->multiply($this->speedMultiplier));
+    }
+
+    public function setSpeedMultiplier(float $multiplier) : void{
```

Comment:
You can't set this. Block objects aren't persistent.

Reply:
It's not reliable under any conditions as the cache for blocks may be destroyed any time. We don't have setters for any other type-level properties like this. Unless this was stored in a blockstate property or tile property, it will just vanish and you'll be none the wiser why it doesn't work.